src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonConverterOfT.cs

```diff
@@ -392,6 +393,6 @@ internal void VerifyWrite(int originalDepth, Utf8JsonWriter writer)
         /// <param name="writer">The <see cref="Utf8JsonWriter"/> to write to.</param>
         /// <param name="value">The value to convert.</param>
         /// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
-        public abstract void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions options);
+        public abstract void Write(Utf8JsonWriter writer, [DisallowNull] T value, JsonSerializerOptions options);
```

Comment:
I am not sure we want to always disallow _null_. We may have "fast path" code flows for internal converters that pass _null_. Also, since this may become a public API there may be cases where someone wants to pass _null_.

Reply:
This API is already public. Having the `JsonSerializer` pass in null or having the user expect/handle null as the input to the `Write` method would be a behavioral breaking change. We can change the annotation if/when we do that.

Also, even if the `JsonSerializer` internally passes null for some reason, we should annotate the public surface area with the intent of the API and what the user can expect when they implement the abstract class `JsonConverter<T>`. What I recall from previous/offline discussions was that we never pass in null to the `Write` method so `DisallowNull` makes sense.

Btw, I added` if (value == null) throw;` to all the `Write` methods within our test converters to validate that we never pass null.

Reply:
Yes I see this is actually on the pubic type.

Reply:
Note that when we make "HandleNullValue" property _protected_ (it is _internal_ now), Write() will get nulls passed in. At that time, I believe should remove the `[DisallowNull]` here